sklearn/cross_validation.py

```diff
@@ -332,6 +332,83 @@ def __len__(self):
         return self.n_folds
 
 
+def disjoint_label_folds(y, n_folds=3):
+    """Creates folds where a same label is not in two different folds
+    
+    Parameters
+    ----------
+    y: iterable of shape (n_samples, )
+        contains an id for each sample
+        The folds are built so that the same id doesn't appear in two different folds
+    
+    n_folds: int, default is 3
+        number of folds to split the data into
+        
+    Returns
+    -------
+    folds: numpy array of shape (n_samples, )
+        array of integers between 0 and (n_folds - 1)
+        folds[i] contains the folds to which sample i is assigned.
+        
+    Notes
+    -----
+    The folds are built by distributing the labels by frequency of appearance.
+    """
+    labels = np.array(y)
+    unique_labels = np.unique(labels)
+    
+    # number of occurrence of each label (its "weight")
+    weight_per_label = sorted([(sum(labels == label), label) for label in unique_labels])
+    # Total weight of each fold
+    weight_per_fold = np.zeros(n_folds)
+    # For each sample, a digit between 0 and (n_folds - 1) to tell which fold it belongs to
+    folds = np.zeros(len(labels))
+    
+    # While there are weights, distribute them
+    # Specifically, add the biggest weight to the lightest fold
+    while weight_per_label:
+        ind_min = np.argmin(weight_per_fold)
+        w, label = weight_per_label.pop()
+        weight_per_fold[ind_min] += w
+        folds[labels == label] = ind_min
+    
+    return folds
+
+
+class DisjointLabelKfold(_BaseKFold):
+    def __init__(self, y, n_folds=3):
+        """Creates K approximately equilibrated folds
+            where the same label will not appear in two different folds
+
+        Parameters
+        ----------
+        y: numpy array of shape (n_samples, )
+            contains an id for each sample
+            The folds are built so that the same label doesn't appear in two different folds
+
+        n_folds: int, default is 3
+            number of folds
```

Comment:
Could you add a nice `Examples` section like we have for `LeavePLabelOut` [here](https://github.com/scikit-learn/scikit-learn/blob/59efcb5/sklearn/cross_validation.py#L537)?